tensorflow/core/graph/mkl_layout_pass.cc

```diff
@@ -3193,35 +3192,44 @@ Status MklLayoutRewritePass::MergeConv2DWithBiasAdd(std::unique_ptr<Graph>* g,
 
   // Incoming data edges from 'pred' node and 'succ' node to new 'new_node'
   // node are already copied in BuildNode. We handle control edges now.
+  std::unordered_set<Node*> unique_node;
   for (const Edge* e : pred->in_edges()) {
     if (e->IsControlEdge()) {
-      if (!DoesControlEdgeExist(e->src(), new_node)) {
+      auto result = unique_node.insert(e->src());
+      if (result.second) {
         (*g)->AddControlEdge(e->src(), new_node, false);
```

Comment:
Since we have made sure not to call `AddControlEdge` with duplicate edges, we can pass `allow_duplicates = true` here to skip the `O(#edges)` check in the routine. Please also add a comment in the code that we pass `true` to skip the expensive check. Same for other places.

Reply:
changed!
thanks!